internal/messaging/processor.go

```diff
@@ -183,9 +230,46 @@ func (p *processor) Respond(msg *Message) error {
 	}
 
 	md := &msg.Metadata
-	// rewrite sender & recipient metadata
-	md.Recipient = md.Sender
-	md.Sender = p.userID
+
+	mybotAddress := md.Recipient
+
+	/// TODO: Uncomment after Chewques implementation
+	// No message should be without cheque?
+	if md.Cheques == nil {
+		return fmt.Errorf("%w: %s", ErrMissingCheques, msg.Type)
+	}
+
+	for i := 0; i < len(md.Cheques); i++ {
+		// Get info from cheque
+		// Check if the TO CM-Account is correct
+		if !p.identificationHandler.isMyCMAccount(md.Cheques[0].ToCMAccount) {
+			return fmt.Errorf("incorrect CMAccount")
+		}
+
+		// Lookup the FROM CM-Account which is in the cheque
+		// Verify that the sending bot is part of this FROM
+		isInCmAccount, err := p.identificationHandler.isBotInCMAccount(md.Sender[1:43], md.Cheques[i].FromCMAccount)
```

Comment:
what is this [1:43] "magical" slice? bot addr from userID?
imo, it would be better if we avoid doing it like this and use more described/ structured approach. e.g, you can use matrix userID type and its methods, I think